### PR TITLE
fix(crux-ui): clear registry fields on type change

### PR DESCRIPTION
**Pull request by peterambrus3**
fix(crux-ui): clear registry fields on type change

fix(crux-ui): google registry url not mapped in edit mode

### CODE REVIEW

**Review by polaroi8d — APPROVED** [review]
LGTM

File: web/crux-ui/src/components/registries/edit-registry-card.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ interface EditRegistryCardProps {
   className?: string
   registry?: RegistryDetails
   onSubmittingChange?: (submitting: boolean) => void
-  onRegistryEdited: (product: Registry) => void
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/edit-registry-card.tsx:37]: lol :D

File: web/crux-ui/src/components/registries/edit-registry-card.tsx [hunk 2]
```diff
@@ -105,6 +105,19 @@ const EditRegistryCard = (props: EditRegistryCardProps) => {
 
   const registryType = formik.values.type
 
+  const handleRegistryTypeChange = (changedRegistry: RegistryType) => {
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/edit-registry-card.tsx:108]: use `onRegistryTypeChange` please instead.

File: web/crux-ui/src/components/registries/edit-registry-card.tsx [hunk 3]
```diff
@@ -105,6 +105,19 @@ const EditRegistryCard = (props: EditRegistryCardProps) => {
...
+    if (registryType !== changedRegistry) {
+      const registryFields = ['imageNamePrefix', 'url', 'token', 'selfManaged', 'user', '_private']
```
Comment (m8vago) [inline web/crux-ui/src/components/registries/edit-registry-card.tsx:110]: @robot9706 Can we replace this with some yup magic? I mean if the type change in the models.ts we will forget this one within a heartbeat.

Comment (robot9706) [reply web/crux-ui/src/components/registries/edit-registry-card.tsx:110]: I was also looking at this yesterday. I think yup can't help in this case, but maybe we could extract fields from fromik and use custom attributes to mark them?

File: web/crux-ui/src/components/registries/edit-registry-card.tsx (hunk 2 above)
Comment (peterambrus3) [reply web/crux-ui/src/components/registries/edit-registry-card.tsx]: done, thx

File: web/crux-ui/src/components/registries/edit-registry-card.tsx (hunk 3 above)
Comment (peterambrus3) [reply web/crux-ui/src/components/registries/edit-registry-card.tsx]: done, can you pls check again?

Comment (robot9706) [reply web/crux-ui/src/components/registries/edit-registry-card.tsx]: Looks cool 😎 maybe the `{reset: true}` object could be moved to a const field other than that LGTM

**Review by robot9706 — APPROVED** [review]
😎 but wait for @m8vago's comments

File: web/crux-ui/src/components/registries/edit-registry-card.tsx (hunk 3 above)
Comment (peterambrus3) [reply web/crux-ui/src/components/registries/edit-registry-card.tsx]: Done, thx

**Review by m8vago — APPROVED** [review]
good job!

**Pull request merged**